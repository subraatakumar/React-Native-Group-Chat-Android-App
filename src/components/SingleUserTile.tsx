import React, {useEffect} from 'react';
import {View, Image, Text, Alert} from 'react-native';
import {Constants, ThemeColors} from '../settings/config';
import CustomButton from './CustomButton';
import man1 from '../assets/img/man1.png';
import people from '../assets/img/people.png';
import {shadows} from '../styles/shadows';
import {SingleUserType} from '../settings/types';
import {
  deleteGroup,
  hideModal,
  showModal,
  useAppDispatch,
} from '../redux/store';
import {CustomModalTypes} from './CustomModal';
import {useSelector} from 'react-redux';

type SingleUserTileProps = {
  bc?: string; // Tile background color
  p?: number | null; // padding
  m?: number | null; // Margin
  br?: number | null; //
  u: SingleUserType;
};

const SingleUserTile = ({
  u,
  bc = ThemeColors.tileBackColor,
  p = 5,
  m = 5,
  br = 20,
}: SingleUserTileProps) => {
  const dispatch = useAppDispatch();

  //console.log('Single User Tile Component u: ', u);
  const {groupDeletionError, groupDeletionStatus} = useSelector(
    (state: any) => state.groupReducer,
  );

  useEffect(() => {
    if (groupDeletionStatus === Constants.LOADING) {
      dispatch(
        showModal({
          type: CustomModalTypes.WARNING,
          title: 'Getting Users',
          body: 'Please wait Getting List Of Users.',
          closable: false,
        }),
      );
      return;
    }

    if (groupDeletionStatus === Constants.REJECTED) {
      dispatch(
        showModal({
          type: CustomModalTypes.ERROR,
          title: 'Getting Users',
          body: groupDeletionError,
          closable: true,
        }),
      );
      return;
    }

    if (groupDeletionStatus == Constants.FULFILLED) {
      dispatch(hideModal());
      return;
    }
  }, [groupDeletionStatus, groupDeletionError]);

  const onSubmit = () => {
    Alert.alert(
      'Deleting A Group',
      "Are you sure? It can't be reverted. Group will all it's messages will be deleted!",
      [
        {
          text: 'Cancel',
          onPress: () => {
            console.log('Cancel Pressed');
          },
          style: 'cancel',
        },
        {
          text: 'OK',
          onPress: () => {
            dispatch(deleteGroup(u));
            console.log('Ok Pressed');
          },
        },
      ],
    );
  };

  return (
    <View
      style={{
        flexDirection: 'row',
        margin: m,
        padding: p,
        backgroundColor: bc,
        borderRadius: br,
        ...shadows,
      }}>
      <Image
        source={u.isGroup ? people : man1}
        style={{width: 40, height: 40, borderRadius: 20}}
      />
      <View style={{marginLeft: 10, flex: 1}}>
        <Text style={{fontSize: 14, fontWeight: 'bold', color: '#000'}}>
          {u && u.name}
        </Text>
        <Text style={{fontSize: 12}}>Latest Message</Text>
      </View>
      {u.isGroup && (
        <CustomButton
          leftIcon="trash"
          bgc={ThemeColors.primary}
          onPressFn={onSubmit}
        />
      )}
    </View>
  );
};

export default SingleUserTile;
