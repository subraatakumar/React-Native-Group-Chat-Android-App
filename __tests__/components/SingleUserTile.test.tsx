import React from 'react';
import renderer from 'react-test-renderer';
import SingleUserTile from '../../src/components/SingleUserTile';

test('', () => {
  const singleUserTile = renderer.create(
    <SingleUserTile
      u={{
        uid: '',
        name: '',
        email: null,
        isGroup: false,
        members: null,
        adminId: null,
        createdAt: null,
      }}
    />,
  );
  expect(singleUserTile.toJSON()).toMatchSnapshot();
});
