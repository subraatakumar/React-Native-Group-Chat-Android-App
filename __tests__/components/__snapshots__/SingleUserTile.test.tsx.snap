// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 20,
      "flexDirection": "row",
      "margin": 5,
      "padding": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.7,
      "shadowRadius": 3,
    }
  }
>
  <Image
    source="test-file-stub"
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Latest Message
    </Text>
  </View>
</View>
`;
